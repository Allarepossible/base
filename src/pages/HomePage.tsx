import React, {useEffect} from 'react';
import {connect} from 'react-redux';

import ContactInfo from 'containers/ContactInfo';
import {fetchContact} from 'actions/contact';
import {State} from 'entities/info';
import {ContactId} from 'entities/contact';
import Page from 'pages/Page';

interface Props {
    title: string;
    contactId: ContactId;
    getContact: (p: number) => void;
}

const HomePage = ({title, contactId, getContact}: Props) => {
    useEffect(() => {
        if (contactId) {
            getContact(contactId);
        }
    }, [contactId, getContact]);

    return (
        <Page title={title}>
            {contactId && <ContactInfo contactId={contactId}/>}
        </Page>
    );
};

export const mapStateToProps = ({info}: State) => {
    const {name, contactId} = info.data || {};

    return {title: name, contactId};
};

const mapDispatchToProps = {
    getContact: fetchContact,
};

export default connect(mapStateToProps, mapDispatchToProps)(HomePage);
