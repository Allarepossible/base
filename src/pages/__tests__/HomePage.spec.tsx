import React from 'react';
import {HashRouter as Router, Route} from 'react-router-dom';
import {shallow} from 'enzyme';

import {Info} from 'entities/info';

import HomePage, {mapStateToProps} from '../HomePage';

const info: Info = {
    name: 'MEDSI',
    contactId: 1,
};

it('Renders Home Page', () => {
    const result = shallow(<Router><Route exact path="/" component={HomePage} /></Router>);

    expect(result).toMatchSnapshot();
});

describe('Home Page', () => {
    it('mapStateToProps() select info, if it is in state', () => {
        const state = {
            info: {
                data: info,
            },
        };
        const expectedResult = {
            title: 'MEDSI',
            contactId: 1,
        };

        expect(mapStateToProps(state)).toEqual(expectedResult);
    });
});
