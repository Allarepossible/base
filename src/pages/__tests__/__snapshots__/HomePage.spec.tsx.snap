// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Home Page 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(HomePage)",
        "type": [Function],
      }
    }
    exact={true}
    path="/"
  />
</Router>
`;
